src/browser/bounds_changed_state_tracker.ts

```diff
@@ -281,11 +284,29 @@ export default class BoundsChangedStateTracker {
         : true;
 
     //tslint:disable-next-line
-    private handleBoundsChange = (isAdditionalChangeExpected: boolean, force = false): boolean => {
+    private handleBoundsChange = (isAdditionalChangeExpected: boolean, force = false, bounds: Rectangle = null): boolean => {
 
         let dispatchedChange = false;
 
-        const currentBounds = this.getCurrentBounds();
+        let currentBounds;
+        if (bounds) {
+            let windowState = WindowState.Normal;
+            if (this.browserWindow.isMaximized()) {
+                windowState = WindowState.Maximized;
+            }
+            if (this.browserWindow.isMinimized()) {
+                windowState = WindowState.Minimized;
+            }
+            const frame = this.browserWindow._options.frame;
+
+            currentBounds = {
+              ...bounds,
+              frame,
+              windowState
+            };
```

Comment:
instead of duplicating this.getCurrentBounds here, easier to call this.getCurrentBounds() and override (x,y) from bounds ?